modules/features/search/src/main/java/au/com/shiftyjelly/pocketcasts/search/SearchResultsPage.kt

```diff
@@ -107,10 +121,65 @@ fun SearchResultsPage(
 }
 
 @Composable
-private fun SearchResultsView() {
-    Column {
-        SearchResultsHeaderView(title = stringResource(LR.string.podcasts))
-        SearchResultsHeaderView(title = stringResource(LR.string.episodes))
+private fun SearchResultsView(
+    state: SearchState.Results,
+    onPodcastClick: (Podcast) -> Unit,
+    onFolderClick: (Folder, List<Podcast>) -> Unit,
+    onScroll: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val nestedScrollConnection = remember {
+        object : NestedScrollConnection {
+            override suspend fun onPostFling(consumed: Velocity, available: Velocity): Velocity {
+                onScroll()
+                return super.onPostFling(consumed, available)
+            }
+        }
+    }
+    LazyColumn(
+        modifier = modifier
+            .nestedScroll(nestedScrollConnection)
+    ) {
+        item { SearchResultsHeaderView(title = stringResource(LR.string.podcasts)) }
+        item {
+            LazyRow(contentPadding = PaddingValues(horizontal = 8.dp),) {
+                items(
+                    items = state.podcasts,
+                    key = { it.adapterId }
+                ) { folderItem ->
+                    when (folderItem) {
+                        is FolderItem.Folder -> {
+                            SearchFolderItem(
+                                folder = folderItem.folder,
+                                podcasts = folderItem.podcasts,
+                                onClick = { onFolderClick(folderItem.folder, folderItem.podcasts) }
+                            )
+                        }
+
+                        is FolderItem.Podcast -> {
+                            SearchPodcastItem(
+                                podcast = folderItem.podcast,
+                                onClick = { onPodcastClick(folderItem.podcast) },
+                            )
+                        }
+                    }
+                }
+            }
+            HorizontalDivider(
+                startIndent = 16.dp,
+                modifier = modifier.padding(top = 20.dp, bottom = 4.dp)
+
+            )
+        }
+        item { SearchResultsHeaderView(title = stringResource(LR.string.episodes)) }
+        state.episodes.forEach {
+            item {
+                SearchEpisodeItem(
+                    episode = it,
+                    onClick = {},
+                )
+            }
```

Comment:
Do we want to use `items` here instead of `forEach { item {} }`? If not, I think we could replace the `LazyColumn` with a `Column`.

Reply:
Oh yes, used items in c8f7cab. Thanks for pointing.